Check/Database/Fragmentation.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * @file
+ * Contains \SiteAudit\Check\Database\Fragmentation.
+ */
+
+/**
+ * Class SiteAuditCheckDatabaseFragmentation.
+ */
+class SiteAuditCheckDatabaseFragmentation extends SiteAuditCheckAbstract {
+  /**
+   * Implements \SiteAudit\Check\Abstract\getLabel().
+   */
+  public function getLabel() {
+    return dt('Database Fragmentation');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getDescription().
+   */
+  public function getDescription() {
+    return dt("Reports all the tables with fragmentation greater than 5%");
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultFail().
+   */
+  public function getResultFail() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultInfo().
+   */
+  public function getResultInfo() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultPass().
+   */
+  public function getResultPass() {
+    return dt('None of the tables has fragmentation of more than 5%');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultWarn().
+   */
+  public function getResultWarn() {
+    arsort($this->registry['database_fragmentation']);
+    if (drush_get_option('html')) {
+      $ret_val = '<table class="table table-condensed">';
+      $ret_val .= '<thead><tr><th>' . dt('Table Name') . '</th><th>' . dt('Fragmentation Percentage') . '</th></tr></thead>';
+      $ret_val .= '<tbody>';
+      foreach ($this->registry['database_fragmentation'] as $name => $ratio) {
+        $ret_val .= '<tr>';
+        $ret_val .= '<td>' . $name . '</td>';
+        $ret_val .= '<td>' . $ratio . '</td>';
+        $ret_val .= '</tr>';
+      }
+      $ret_val .= '</tbody>';
+      $ret_val .= '</table>';
+    }
+    else {
+      $ret_val  = dt('Table Name: Fragmentation Percentage') . PHP_EOL;
+      if (!drush_get_option('json')) {
+        $ret_val .= str_repeat(' ', 4);
+      }
+      $ret_val .= '---------------------';
+      foreach ($this->registry['database_fragmentation'] as $name => $ratio) {
+        $ret_val .= PHP_EOL;
+        if (!drush_get_option('json')) {
+          $ret_val .= str_repeat(' ', 4);
+        }
+        $ret_val .= "$name: $ratio";
+      }
+    }
+    return $ret_val;
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {
+    if ($this->getScore() == SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_WARN) {
+      return dt('Run "OPTIMIZE TABLE" on the fragmented tables. Refer to https://dev.mysql.com/doc/refman/5.5/en/optimize-table.html for more details.');
```

Comment:
Use https://dev.mysql.com/doc/en/optimize-table.html as it's evergreen and will automatically redirect to current stable release.